src/System.CommandLine.Suggest.Tests/SuggestionDispatcherTests.cs

```diff
@@ -22,15 +21,54 @@ private static RegistrationPair CurrentExeRegistrationPair()
         private static string CurrentExeFullPath() => Path.GetFullPath(_currentExeName);
 
         [Fact]
-        public async Task InvokeAsync_executes_suggestion_command_for_executable()
+        public async Task InvokeAsync_executes_registered_executable()
         {
+            string receivedTargetExeName = null;
+
             string[] args = $@"get -p 12 -e ""{CurrentExeFullPath()}"" -- {_currentExeName} add".Tokenize().ToArray();
 
-            var suggestions = await InvokeAsync(args, new TestSuggestionRegistration(CurrentExeRegistrationPair()));
+            var suggestions = await InvokeAsync(
+                                  args,
+                                  new TestSuggestionRegistration(CurrentExeRegistrationPair()),
+                                  new AnonymousSuggestionStore(
+                                      (targetExeName, targetExeArgs, _) =>
+                                      {
+                                          receivedTargetExeName = targetExeName;
 
-            suggestions
-                .Should()
-                .Be($"package{Environment.NewLine}reference{Environment.NewLine}");
+                                          return "";
+                                      }));
+
+            receivedTargetExeName.Should().Be(CurrentExeFullPath());
+
+            
+        }
+
+        [Fact]
+        public async Task InvokeAsync_executes_suggestion_command_for_executable()
+        {
+            string receivedTargetExeArgs = null;
+
+            var args = PrepareArgs($@"get -p 12 -e ""{CurrentExeFullPath()}"" -- {_currentExeName} add");
```

Comment:
Use deconstruction here? Saves on the weird lookings `args.args` below.
```suggestion
            var (args, _) = PrepareArgs($@"get -p 12 -e ""{CurrentExeFullPath()}"" -- {_currentExeName} add");
```